embc-app/ClientApp/src/app/shared/components/date-time-picker/date-time-picker.component.ts

```diff
@@ -0,0 +1,164 @@
+import { Component, OnInit, Input, Output, EventEmitter, forwardRef } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+import { NgbTimeStruct } from '@ng-bootstrap/ng-bootstrap';
+import * as moment from 'moment';
+
+// tslint:disable: no-use-before-declare
+export const DATETIMEPICKER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => DateTimePickerComponent),
+  multi: true
+};
+
+interface DateTimeStruct {
+  dateString: string;
+  hour: number;
+  minute: number;
+  second: number;
+}
+
+@Component({
+  selector: 'app-date-time-picker',
+  templateUrl: './date-time-picker.component.html',
+  styleUrls: ['./date-time-picker.component.scss'],
+  providers: [DATETIMEPICKER_VALUE_ACCESSOR]
+})
+export class DateTimePickerComponent implements OnInit, ControlValueAccessor {
+
+  @Input() disabled = false;
+  @Input() value: Date = null;
+  @Output() valueChange = new EventEmitter<Date>();
+
+  model: DateTimeStruct = {
+    dateString: null,
+    hour: 0,
+    minute: 0,
+    second: 0
+  };
+
+  dateString: string = null;
+  time: NgbTimeStruct = null;
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
+  // ControlValueAccessor methods follow...
+
+  // This is a placeholder method that we use to emit changes back to the form
+  onChange = (_: any) => { };
+  onTouched = () => { };
+
+  // This is a basic setter that the Angular forms API is going to use
+  writeValue(value: any): void {
+    const structValue = this.fromJSDate(value);
+    this.setModel(structValue);
+  }
+
+  // Allows Angular to register a function to call when the model changes.
+  // This is how we emit the changes back to the form
+  registerOnChange(fn: any): void { this.onChange = fn; }
+
+  registerOnTouched(fn: any): void { this.onTouched = fn; }
+
+  // Allows Angular to disable this component.
+  setDisabledState(isDisabled: boolean): void { this.disabled = isDisabled; }
+
+  // end of ControlValueAccessor
+
+  isValid(model: DateTimeStruct): boolean {
+    if (model.dateString) {
+      const d = moment(model.dateString, 'YYYY-MM-DD', true);
+      return d.isValid();
+    }
+    return false;
+  }
+
+  updateDate(newValue: string): void {
+    this.setModel({ ...this.model, dateString: newValue });
+    this.propagateModelChange();
+  }
+
+  updateTime(newValue: NgbTimeStruct): void {
+    this.setModel({ ...this.model, ...newValue });
+    this.propagateModelChange();
+  }
+
+  formatDate(date: Date): string {
+    if (!date) {
+      return null;
+    }
+    const year = date.getFullYear();
+    const month = date.getMonth() + 1;
+    const day = date.getDate();
+    return `${this.pad(year)}-${this.pad(month)}-${this.pad(day)}`;
+  }
+
+  formatTime(date: Date): NgbTimeStruct {
```

Comment:
Suggest renaming tihs to 'dateToNgbTimeStruct'.

Reply:
Agree, will rename